File: histogram/histogram.js
const fs = require("fs");

fs.readFile("input.txt", "utf-8", (err, data) => {
inputData = data.toString().split("\n").join("").split(" ").join("").split("");

let height = 0;
const dictionary = inputData.reduce((acc, i) => {
	if (acc.hasOwnProperty(i)) {
		acc[i] += 1;

	} else {
		acc[i] = 1;
	}        
    if (height < acc[i]) {
        height = acc[i];
    }
	return acc;
}, {});

let position = Object.keys(dictionary).sort((a,b) => a.charCodeAt(0)-b.charCodeAt(0));

let result = '';
for (let i = height; i > 0; i--) {
	let row = '';
	for (let j = 0; j < position.length; j++) {
		if (i > dictionary[position[j]]) {
			row += ' ';
		} else {
			row += '#';
		}
	}
  
	row += '\n';
	result += row;
}

result += position.join('');

fs.writeFile("output.txt", result += '\n', (err) => {})

});

File: histogram/histogram.py
def printchart(s):
    dict = {}
    maxsymcount = 0
    for sym in s:
        if sym not in dict:
            dict[sym]=0
        dict[sym]+=1
        maxsymcount=max(maxsymcount, dict[sym])
    sortseduniqsyms=sort(dict.keys())
    for row in range(maxsymcount, 0, -1):
        for sym in sortseduniqsyms:
            if dict[sym] >= row:
                print('#', end='')
            else:
                print(' ', end='')
        print()
    print(''.join(sortseduniqsyms))
